org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLServerSocketLauncher.java

```diff
@@ -0,0 +1,62 @@
+package org.eclipse.lsp4xml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.nio.channels.AsynchronousServerSocketChannel;
+import java.nio.channels.AsynchronousSocketChannel;
+import java.nio.channels.Channels;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.eclipse.lsp4j.jsonrpc.Launcher;
+import org.eclipse.lsp4j.jsonrpc.MessageConsumer;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+import com.google.common.base.Objects;
+
+public class XMLServerSocketLauncher {
+
+	private static final int DEFAULT_PORT = 5_008;
+
+	public static void main(String[] args) {
+		new XMLServerSocketLauncher().run(args);
+	}
+
+	public void run(String[] args) {
+		try {
+			AsynchronousServerSocketChannel _open = AsynchronousServerSocketChannel.open();
+			int _port = getPort(args);
```

Comment:
if parsing the port fails, the exception will be swallowed (just logged). Program should end with exit code != 0